src/jit/morph.cpp

```diff
@@ -5630,7 +5630,8 @@ GenTreePtr Compiler::fgMorphArrayIndex(GenTreePtr tree)
         // dereference.
         // Also we allocate the temporary when the arrRef is sufficiently complex/expensive.
         //
-        if ((arrRef->gtFlags & (GTF_ASG | GTF_CALL | GTF_GLOB_REF)) || gtComplexityExceeds(&arrRef, MAX_ARR_COMPLEXITY))
+        if ((arrRef->gtFlags & (GTF_ASG | GTF_CALL | GTF_GLOB_REF)) ||
```

Comment:
Update the comment to add "or when the arrRef is a GT_FIELD"
Is it actually due to the complexity of a GT_FIELD expansion or for some other reason?